bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/util/TestUtil.java

```diff
@@ -0,0 +1,37 @@
+package com.google.cloud.bigtable.util;
+
+import com.google.cloud.bigtable.config.Logger;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+
+public class TestUtil {
+  private static final Logger LOG = new Logger(TestUtil.class);
+
+  public static Server buildServer(ServerBuilderFactory serverBuilderFactory) throws IOException {
+    int maxRetries = 3;
+
+    for (int retry = 0; retry < maxRetries; retry++) {
+      int port;
+      try (ServerSocket s = new ServerSocket(0)) {
+        port = s.getLocalPort();
+      }
+      Server server = serverBuilderFactory.buildServer(port).build();
+      try {
+        server.start();
+        return server;
+      } catch (IOException e) {
+        LOG.trace("Could not setup port; retrying ...", e);
+      }
+    }
+
+    // Exhausted retries.
+    throw new IllegalArgumentException("Could not create server after " + maxRetries + " retries; unable to find port");
+  }
+
+  public interface ServerBuilderFactory {
```

Comment:
We're on Java 7, so I guess it's a functional interface and abstract classes on the calling side for now ... I'm open to better suggestions